commands/Tools/Step 3 - Modify schematisation/control_structures.py

```diff
@@ -0,0 +1,210 @@
+# -*- coding: utf-8 -*-
+# (c) Nelen & Schuurmans, see LICENSE.rst.
+
+import logging
+
+from PyQt4.QtCore import Qt
+
+
+from ThreeDiToolbox.views.control_structures_dockwidget import ControlStructuresDockWidget  # noqa
+from ThreeDiToolbox.commands.base.custom_command import CustomCommandBase
+from ThreeDiToolbox.utils.threedi_database import get_databases
+from ThreeDiToolbox.threedi_schema_edits.controlled_structures import \
+    ControlledStructures
+
+log = logging.getLogger(__name__)
+
+ACTIVE_TAB = " (active tab)"
+TABLE_CONTROL = 'table_control'
+BOTTOM_UP = 'Bottom up'
+RULE_OPERATOR_BOTTOM_UP = '>'
+RULE_OPERATOR_TOP_DOWN = '<'
+V2_WEIR_VIEW_TABLE = "v2_weir_view"
+V2_WEIR_TABLE = 'v2_weir'
+INITIAL_WATERLEVEL = "initial_waterlevel"
+WATERLEVEL = 'waterlevel'
+ACTION_TYPE_SET_CREST_LEVEL = 'set_crest_level'
+
+
+class CustomCommand(CustomCommandBase):
+    """
+    command to that will load and start an edit session for the connected
+    point layer and verify the data added to that layer
+    """
+
+    def __init__(self, *args, **kwargs):
+        self.args = args
+        self.kwargs = kwargs
+        self.iface = kwargs.get('iface')
+        self.ts_datasource = kwargs.get('ts_datasource')
+        self.dockwidget_controlled_structures = None
+        self.control_structure = None
+
+        self.databases = get_databases()
+        # Remove u'selected' spatialite and postgresdatabases
+        # from self.databases to prevent confusion about which database
+        # is meant by it
+        if 'spatialite: ' in self.databases:
+            self.databases.pop('spatialite: ', None)
+        if 'postgres: None' in self.databases:
+            self.databases.pop('postgres: None', None)
+
+    def run(self):
+        self.show_gui()
+        self.dockwidget_controlled_structures.combobox_input_model\
+            .activated.connect(self.update_dockwidget_ids)
+        self.dockwidget_controlled_structures.pushbutton_input_save\
+            .clicked.connect(self.save_control_structure)
+
+    def run_it(self):
+        """Run it."""
```

Comment:
I think it could be removed. 

It would be doing the same as the save_control_structure(self) function,
but save_control_structure(self) is more comprehensible.